File: 0141-linked-list-cycle/0141-linked-list-cycle.java
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public boolean hasCycle(ListNode head) {
        ListNode fast = head;
        ListNode slow = head;
        // Edge Case:
        if(head == null || head.next == null) return false;
        
        while(fast != null && fast.next != null){
            fast = fast.next.next;
            slow = slow.next;
            // Checking for the same node:
            if(fast == slow){
                return true;
            }
        }
        
        return false;
    }
}

File: 0141-linked-list-cycle/0141-linked-list-cycle.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} head
 * @return {boolean}
 */
var hasCycle = function(head) {
    let fast = head;
    let slow = head;
    
    // Edge Case:
    if(head == null || head.next == null)
        return false;
    
    // Iteration:
    while(fast && fast.next){
        fast = fast.next.next;
        slow = slow.next;
        
        // Checking the Condition:
        if(fast == slow){
            return true;
        }
    }
    return false;
};